### DISCUSSION TITLE
Example document for advanced math

### DISCUSSION THREAD

**Discussion by GuillaumeDehaene** [quarto-dev/quarto-cli#7222] [Q&A]
### Description

Hello everyone,

TLDR:
1. Is there an existing example file for complex math articles ([like this one](https://arxiv.org/pdf/1612.05053.pdf))?
1. If not, I'd like to write one: do you have any recommendations? Are there features that I should highlight in such a example document?

I have recently discovered Quarto and I absolutely love this project (after having discovered Pandoc and absolutely loving that too).

The background of my question is that I am a mathematician that is very frustrated with working with latex. In a nutshell, I am frustrated that latex, besides the math mode, is very cumbersome and hard to understand.
Currently, I use Lyx as a latex editor.
I would like to use Quarto instead.

One thing currently slowing me down is that I cannot figure out from the docs or google the right way to do a lot of the basic operations of building a complex math document.
For example, in order to number equations, I needed to track down [this discussion](https://github.com/quarto-dev/quarto-cli/discussions/4132)
Is there a reference example that I missed during my search?

If not, I would like to contribute to the project by writing:

1. A example file with:

  1. The right way to do basic math operations in Quarto, in order to show equivalence with Latex
  1. A demo of the cool features that Quarto has, in order to show that Quarto is actually better
  
1. Perhaps, if that is useful, a documentation section to guide latex user into porting their workflow into Quarto

This example file should cover the following topics:
1. Latex equivalence:
    - Front matter
    - Macros
    - Using equation, align and other ams environments
    - Using label and eqref for equation numbering
    - Cross references
    - Configuring math-jax for math articles: equation numbers
    - Theorem-like environments
    - Citation
    - Foonotes
    - Appendix
1. Quarto features:
    - using python to build figures
    - code blocks
    - diagrams
    - callout blocks
    - line based annotations
    - citation definition in the preamble
 
Are there any other features that I should add to this list that you think would be useful for this target audience?

Here is an example article that I would like to be writable in Quarto: https://arxiv.org/pdf/1612.05053.pdf

Cheers
Guillaume

**Comment by mcanouil** [comment]
I am not sure what you mean exactly.
There is no need for a template to enable LaTeX packages or MathJax/KaTeX maths.
Math get complex/advanced not because of the template but because of the authors.
Did you take a look at the extension page/section? I believe there is everything you need in there: <https://quarto.org/docs/extensions/>.


Also see how to write Math: <https://quarto.org/docs/authoring/markdown-basics.html#equations>.

**Comment by GuillaumeDehaene** [comment]
No, I do not mean a template to enable anything. I meant it in the sense of an example. I'll clarify the initial post.

**Reply by mcanouil** [reply]
I am not sure what you expect.
The maths are written exactly as in LaTeX, so take any big math equation you can find on the internet and put it as is in Quarto then you'll have your example.


Note that template is not a synonym of example, so if you only want examples you should not use "template".


````qmd
---
title: "Quarto Playground"
format: html
---

This is a playground for Quarto.

## Example 1

$$
\begin{gather*}
a_0=\frac{1}{\pi}\int\limits_{-\pi}^{\pi}f(x)\,\mathrm{d}x\\[6pt]
\begin{split}
a_n=\frac{1}{\pi}\int\limits_{-\pi}^{\pi}f(x)\cos nx\,\mathrm{d}x=\\
=\frac{1}{\pi}\int\limits_{-\pi}^{\pi}x^2\cos nx\,\mathrm{d}x
\end{split}\\[6pt]
\begin{split}
b_n=\frac{1}{\pi}\int\limits_{-\pi}^{\pi}f(x)\sin nx\,\mathrm{d}x=\\
=\frac{1}{\pi}\int\limits_{-\pi}^{\pi}x^2\sin nx\,\mathrm{d}x
\end{split}\\[6pt]
\end{gather*}
$$

## Example 2

$$
\begin{align*}
 &\vdots\\ 
 &=12+7 \int_0^2
  \left(
    -\frac{1}{4}\left(e^{-4t_1}+e^{4t_1-8}\right)
  \right)\,dt_1\\
 &= 12-\frac{7}{4}\int_0^2 \left( e^{-4t_1}+e^{4t_1-8} \right)\,dt_1\\
 &\vdots % 
\end{align*}
$$


## Example 3

$$
\begin{align}
h_{i}^{n} = - \color{red}h_{i-1}^{n+1} \color{black} \left(\frac{\Delta t}{(\Delta x)^2} \left(\frac{h_{i}^{n} + h_{i-1}^{n}}{2} \right)^3 \right) &+ \color{red} h_{i}^{n+1} \color{black} \left (1+  \frac{\Delta t}{(\Delta x)^2} \left(\frac{h_{i}^{n} + h_{i-1}^{n}}{2} \right)^3 + \frac{\Delta t}{(\Delta x)^2} \left(\frac{h_{i+1}^{n} + h_{i}^{n}}{2} \right)^3 \right) \notag \\
- \color{red}h_{i}^{n+1} \color{black}\left(\frac{\Delta t}{(\Delta x)^2} \left(\frac{h_{i}^{n} + h_{i+1}^{n}}{2} \right)^3  \right) \notag
\end{align}
$$

$$
\begin{align}
h_{N-1}^{n} = - \color{red} h_{N-2}^{n+1} \color{black} \left(\frac{\Delta t}{(\Delta x)^2} \left(\frac{h_{N-1}^{n} + h_{N-2}^{n}}{2} \right)^3 \right) + \color{red} h_{N-1}^{n+1} \color{black} \left (1+  \frac{\Delta t}{(\Delta x)^2} \left(\frac{h_{N-1}^{n} + h_{N}^{n}}{2} \right)^3 + \frac{\Delta t}{(\Delta x)^2} \left(\frac{h_{N}^{n} + h_{N-1}^{n}}{2} \right)^3 \right) \notag \\
- \color{red} h_{N}^{n+1} \color{black} \left(\frac{\Delta t}{(\Delta x)^2} \left(\frac{h_{N-1}^{n} + h_{N}^{n}}{2} \right)^3  \right) \notag
\end{align}
$$
````

**Comment by GuillaumeDehaene** [comment]
Hello,

Thank you very much for your help. I have corrected the discussion to remove all mentions of the word template: you are right, I am instead interested in an example.

Interestingly (and I really don't mean to be cheeky by writing this), the document you wrote fails to render to PDF.
The issue is that the .tex file contains blocks like:
```latex
$$
\begin{AMSENV}
...
\end{AMSENV}
$$
```
which trigger errors in the pdf-engine due to MathMode being opened twice.
I think this really shows that having this example would be good for people, like me, coming from tex and needing to use align.

Here is the status of my WIP: https://github.com/GuillaumeDehaene/quarto_math_example/blob/main/minimal.qmd
It's content is formatted mostly based on this answer https://github.com/quarto-dev/quarto-cli/discussions/4132#discussioncomment-4801747 and adding a macro using the `include-in` commands
Is this the right way for a Quarto document to address these needs?

In another thread (https://github.com/quarto-dev/quarto-cli/issues/7253), you've said that I shoud wrap the latex-environments inside a ` ```{=latex} `
I can see that this changes the pandoc processing of the document: rendering to native indicates that a naked environment parses to `RawInline (Format "tex")` whereas a wrapped on parses to `RawBlock (Format "latex")`
Do you have a specific reason to prefer one over the other?

Again, thank you for your insight.
Best regards

**Reply by mcanouil** [reply]
See:
- https://github.com/quarto-dev/quarto-cli/issues/2870#issuecomment-1283035658

````qmd
---
title: "Quarto Playground"
format:
  html: default
  pdf: default
---

This is a playground for Quarto.

## Example 1

\begin{gather*}
a_0=\frac{1}{\pi}\int\limits_{-\pi}^{\pi}f(x)\,\mathrm{d}x\\[6pt]
\begin{split}
a_n=\frac{1}{\pi}\int\limits_{-\pi}^{\pi}f(x)\cos nx\,\mathrm{d}x=\\
=\frac{1}{\pi}\int\limits_{-\pi}^{\pi}x^2\cos nx\,\mathrm{d}x
\end{split}\\[6pt]
\begin{split}
b_n=\frac{1}{\pi}\int\limits_{-\pi}^{\pi}f(x)\sin nx\,\mathrm{d}x=\\
=\frac{1}{\pi}\int\limits_{-\pi}^{\pi}x^2\sin nx\,\mathrm{d}x
\end{split}\\[6pt]
\end{gather*}

## Example 2

\begin{align*}
 &\vdots\\ 
 &=12+7 \int_0^2
  \left(
    -\frac{1}{4}\left(e^{-4t_1}+e^{4t_1-8}\right)
  \right)\,dt_1\\
 &= 12-\frac{7}{4}\int_0^2 \left( e^{-4t_1}+e^{4t_1-8} \right)\,dt_1\\
 &\vdots % 
\end{align*}


## Example 3

\begin{align}
h_{i}^{n} = - \color{red}h_{i-1}^{n+1} \color{black} \left(\frac{\Delta t}{(\Delta x)^2} \left(\frac{h_{i}^{n} + h_{i-1}^{n}}{2} \right)^3 \right) &+ \color{red} h_{i}^{n+1} \color{black} \left (1+  \frac{\Delta t}{(\Delta x)^2} \left(\frac{h_{i}^{n} + h_{i-1}^{n}}{2} \right)^3 + \frac{\Delta t}{(\Delta x)^2} \left(\frac{h_{i+1}^{n} + h_{i}^{n}}{2} \right)^3 \right) \notag \\
- \color{red}h_{i}^{n+1} \color{black}\left(\frac{\Delta t}{(\Delta x)^2} \left(\frac{h_{i}^{n} + h_{i+1}^{n}}{2} \right)^3  \right) \notag
\end{align}


\begin{align}
h_{N-1}^{n} = - \color{red} h_{N-2}^{n+1} \color{black} \left(\frac{\Delta t}{(\Delta x)^2} \left(\frac{h_{N-1}^{n} + h_{N-2}^{n}}{2} \right)^3 \right) + \color{red} h_{N-1}^{n+1} \color{black} \left (1+  \frac{\Delta t}{(\Delta x)^2} \left(\frac{h_{N-1}^{n} + h_{N}^{n}}{2} \right)^3 + \frac{\Delta t}{(\Delta x)^2} \left(\frac{h_{N}^{n} + h_{N-1}^{n}}{2} \right)^3 \right) \notag \\
- \color{red} h_{N}^{n+1} \color{black} \left(\frac{\Delta t}{(\Delta x)^2} \left(\frac{h_{N-1}^{n} + h_{N}^{n}}{2} \right)^3  \right) \notag
\end{align}

````

**Reply by mcanouil** [reply]
> Do you have a specific reason to prefer one over the other?

It is use case dependent.

Also feel free to submit a PR to <https://github.com/quarto-dev/quarto-examples>

**Reply by GuillaumeDehaene** [reply]
Thank you for your help. I'll gather some feedback from some friends who are experienced with latex and curious about Quarto, and submit a PR once the example is done.
